front/src/modules/ui/dropdown/hooks/useDropdownCloseEffect.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react';
+
+import { useDropdownButton } from './useDropdownButton';
+
+export const UseDropdownCloseEffect = ({
```

Comment:
Can you keep this as a component: DropdownCloseEffect and not as a hook, and instantiate it in DropdownButton JSX?
The idea is to put this useEffect within a child of DropdownButton ; this way when DropdownCloseEffect re-renders because it's listening to isDropdownButtonOpen, the DropdownButton itself does not re-render, which is more optimized!